src/rulesets/Base/base.jl

```diff
@@ -161,3 +161,15 @@ end
 @scalar_rule round(x) zero(x)
 @scalar_rule floor(x) zero(x)
 @scalar_rule ceil(x) zero(x)
+
+# note: rules for ^ are defined in the fastmath_able.jl
+function frule((_, _, Δx, _), ::typeof(Base.literal_pow), ::typeof(^), x::Real, pv::Val{p}) where p
+    y = Base.literal_pow(^, x, pv)
+    return y, (p * y / x * Δx)
+end
+
+function rrule(::typeof(Base.literal_pow), ::typeof(^), x::Real, pv::Val{p}) where p
+    y = Base.literal_pow(^, x, pv)
+    literal_pow_pullback(dy) = NoTangent(), NoTangent(), (p * y / x * dy), NoTangent()
+    return y, literal_pow_pullback
+end
```

Comment:
These are identical to another PR right? Is git / github just confused about what's new and what's not?

https://github.com/JuliaDiff/ChainRules.jl/pull/464/files#diff-dbe52c217d193026d8825d7b34db346dc5c8e7dbab6e28453a03b62b3c196859R164-R175

Reply:
Yes, these are pieces from 2 other PRs, perhaps the result of a poor combination of rebases and merges with changes suggested via Github UI. The codebase shouldn't be affected, but the git history may look a bit weird, so I can simply create a new clean PR (although it will destroy authorship of the suggested edits).

Reply:
No that sounds like a lot of work. I'm 99% sure it's going to merge smoothly & not duplicate things.